Automattic/woocommerce-payments
# Express Checkout separator - honor express checkout button settings

#### Changes proposed in this Pull Request

This is a bit of an addendum to #5399. Shortly after merging #5399, @timur27 and I noticed that the button location/page settings weren't being honored. In some scenarios, the `- or -` separator was displayed even when no button was displayed. This PR corrects that by utilizing the same `should_show_*` checks used by the buttons. 







#### Testing instructions

1. Enable 2 or more express checkout methods(Link, WooPay, Payment Request). The WooPay express checkout button can be enabled via the latest version of the WCPay Dev Tools plugin. Configure the express button by selecting **Customize** under the WooPay Express Checkout settings at **Payments  > Settings**. 
2. Visit a product page where only 1 **- OR -** separator should appear.
3. Add a product to the cart and visit the cart page where only 1 **- OR -** separator should appear. 
4. Visit the cart page where only 1 **- OR -** separator should appear.
5. Using the settings found for each express checkout button under its **Customize** option, disable the button on some pages and check that settings are being honored.
